src/Microsoft.ML.FastTree/Training/Test.cs

```diff
@@ -207,8 +207,8 @@ public class TestHistory : Test
         protected IList<TestResult[]> History;
         protected int Iteration { get; private set; }
 
-        public TestResult BestResult { get; protected internal set; }
-        public int BestIteration { get; protected internal set; }
+        public TestResult BestResult { get; private protected set; }
```

Comment:
Can these just be `protected`?  It seems odd that only inheriting classes from in this assembly can set these properties, but inheriting classes outside can't.

Maybe changing the constructor to `internal` would be better if we don't want external derived classes?

Reply:
Yeah... this entire assembly needs some serious work on what is public and internal. It isn't very consistent. But we can certainly make this small step.